Automattic/simplenote-android
# Implement RecyclerView for tags list

### Fix
This is a partial fix for #249.
I have implemented RecyclerView for tags list. This provides delete animation, enhances the user experience, and makes the tag list more efficient by implementing ViewHolder pattern and RecyclerView's improved performance.

Before | After
-- | --
![before](https://user-images.githubusercontent.com/5239138/60689658-6262ec80-9e85-11e9-86c9-3b299027b8bb.gif)|![after](https://user-images.githubusercontent.com/5239138/60689660-668f0a00-9e85-11e9-9f2a-647ac25296fe.gif)

### Test
1. Go to tags list
2. Edit some tags
3. Notice change animation
3. Delete some tags with and without notes
4. Notice delete animation

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.
